### ISSUE TITLE
Rule S2115: FP when implicit authentication is used

### ISSUE THREAD

**Issue by pierre-loup-tristant-sonarsource** [SonarSource/sonar-dotnet#4304]
### Description

In [security-expected-issues:NetFwk48MVC/Controllers/RSPEC_2115/Compliant/Web.config](https://github.com/SonarSource/security-expected-issues/blob/master/dotnet/NetFwk48MVC/Controllers/RSPEC_2115/Compliant/Web.config) issues are raised in compliant code examples. No issue should be raised when a empty password is located in a connection string that also use implicit authentication.

In general, implicit authentication materialize in the connection string by using the parameter `Trusted_Connection=True`. There is many variates for setting this parameter, the following are all compliant examples:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
	<connectionStrings>
		<add name="CompliantNoPassword" connectionString="Server=myServerAddress;Database=myDataBase;Trusted_Connection=True" />
		<add name="CompliantTrustedConnectionTrue" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Trusted Connection=True" />
		<add name="CompliantIntegratedSecurityTrue" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Integrated Security=true" />
		<add name="CompliantTrustedConnectionYes" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Trusted Connection=Yes" />
		<add name="CompliantIntegratedSecurityYes" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Integrated Security=yes" />
		<add name="CompliantIntegratedSecuritySSPI" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Integrated Security=SSPI" />
		<add name="CompliantTrusted_ConnectionTrue" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Trusted_Connection=True" />
		<add name="CompliantIntegrated_SecurityTrue" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Integrated_Security=true" />
		<add name="CompliantTrusted_ConnectionYes" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Trusted_Connection=Yes" />
		<add name="CompliantIntegrated_SecurityYes" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Integrated_Security=yes" />
		<add name="CompliantIntegrated_SecuritySSPI" connectionString="Server=myServerAddress;Database=myDataBase;Password=;Integrated_Security=sspi" />
		<add name="CompliantForS2115_NoncompliantForS2068" connectionString="Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password=myPassword" />
	</connectionStrings>
</configuration>
```

The following pattern should match all the compliant cases defined above.
`((integrated[_\s]security)|(trusted[_\s]connection))=(sspi|yes|true)` (case insensitive)
